File: apps/web/src/lib/httpConstants.ts
// File : /team-samsara/apps/web/src/lib/httpConstants.ts
// Version : 1.0.0
// Latest commit: feature/string-magic-value-conventions
// Author : Gerrah
// Purpose : HTTP protocol values our code depends on matching exactly. Mirrors the
// backend's HttpConstants.cs - CorrelationIdHeaderName in particular must match the
// backend's CorrelationIdMiddleware.HeaderName exactly, or correlation IDs silently stop
// linking frontend and backend logs together.

export const HttpConstants = {
  correlationIdHeaderName: "X-Correlation-Id",
  bearerPrefix: "Bearer ",
} as const;

File: apps/api/src/TeamSamsara.Shared/Http/HttpConstants.cs
// File : /team-samsara/apps/api/src/TeamSamsara.Shared/Http/HttpConstants.cs
// Version : 1.0.0
// Latest commit: feature/string-magic-value-conventions
// Author : Gerrah
// Purpose : HTTP protocol values our code depends on matching exactly, not covered by any
// typed API in the frameworks we use.

namespace TeamSamsara.Shared.Http;

public static class HttpConstants
{
    #region Fields

    public const string BearerPrefix = "Bearer ";
    public const string ProblemJsonContentType = "application/problem+json";

    #endregion
}
